tools/nuget/README.md

```diff
@@ -0,0 +1,42 @@
+# NuGet package creation process
+
+This directory contains NuGet specification files (.nuspec).
+
+Creating nuget packages:
+
+1. Set `PackageVersion` environment variable that defines the package version.
+
+```console
+set PackageVersion=0.2.0
+```
+
+If environment variable is not set, then the version is automatically obtained from git tag:
+
+```console
+git describe --tags
+```
+
+2. Run `tools/build-nuget.cmd` script to create the package(s). Packages are copied to `.\packages` directory.
+
+3. Push selected package to nuget feed.
+
+Learn more about native code NuGet packages [here](https://docs.microsoft.com/en-us/nuget/guides/native-packages).
+
+## Proposed NuGet package flavors
+
+This section is a `DRAFT` - Work-in-Progress. The following packages MAY be created and published at nuget.org :
+
+| Package Name                 | Desription                                                  | Notes |
+|------------------------------|-------------------------------------------------------------|-------|
+| OpenTelemetry.Cpp.Source     | Complete recursive source code snapshot with dependencies.  |       |
+| OpenTelemetry.Cpp.Dev        | Source code snapshot excluding 3rd party dependencies.      |       |
+| OpenTelemetry.Cpp.Api        | API headers package.                                        |       |
+| OpenTelemetry.Cpp.Sdk        | SDK headers and source package.                             |       |
```

Comment:
Would it make sense to add package dependencies in this table - say `OpenTelemetry.Cpp.Sdk` would also need `OpenTelemetry.Cpp.Api`, `OpenTelemetry.Cpp.Exporters` would need `OpenTelemetry.Cpp.Sdk`  and `OpenTelemetry.Cpp.Ext`.

Reply:
No, I would like to avoid that. I considered that and eventually it was my conscious decision to avoid that. Logical reasoning: dependency management is more predictable for C#. Less predictable for native code. Certain systems simply unpack the nuget (as zip) without dependency management at all. Thus, adding dependency management forces someone to manually add multiple packages, one-by-one.

Thus, I'd rather have us a complete source package:
- **Source**
- **Dev** - stripped-down version, i.e. installing 3rd party deps using alternate mechanism.

For **Sdk** vs **Api**, one can take the **Api** package and not take the **Sdk** package, and should still be able to build the SDK. Thus, adding dependencies to **Api** probably does not make sense.

The reverse-dependency ( **Sdk** depends on **Api** ), can be manually-managed (or at least in a system called CoreXT it would have to be manually managed0. We can discuss this automatic dependency management later on. I am not adding any spec at the moment for either of these packages. Also the other logical reasoning for manual dependency management is if you want to mix and match **Sdk-v3.1.0** with **Api-v3.2.0** , for example. If someone needs this managed automatically, they themselves can create their own metapackage. Maybe **OpenTelemetry.Cpp** should be the metapackage that automatically manages dependencies, but no individual mapping between sub-packages.

Reply:
To recap - maybe just one higher-order _metapackage_ `OpenTelemetry.Cpp` with no suffix, that aggregates all packages and only that one performs dependency management. But no dependency management between the children, i.e. one can still mix-and-match the compatible packages individually by force-installing `OpenTelemetry.Cpp.Sdk-vX` and `OpenTelemetry.Cpp.Api-vY`, **as long as developer knows what they are doing**.